### PR TITLE
[Do Not Merge] Minor refactor LogRecord

### PR DESCRIPTION
**Pull request by reyang**
Currently `LogRecord` requires most of the fields to be initialized during ctor.
This change makes it possible to modify the LogRecord, which in turn makes it possible to pool the LogRecord objects (which could potentially help to reduce allocation).

This PR does not cover the actual pooling logic, it does not change the existing functionality, it does not introduce noticeable changes to perf.

### CODE REVIEW

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Comment by reyang** [comment]
"Integration Tests / redis-test" had a transient failure which seems to be related to container registry:

```text
Step 2/12 : FROM mcr.microsoft.com/dotnet/sdk:6.0-focal AS build
Get "https://mcr.microsoft.com/v2/": context deadline exceeded
Service 'tests' failed to build : Build failed
Error: Process completed with exit code 1.
```

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry/Logs/OpenTelemetryLogger.cs [hunk 1]
```diff
@@ -40,31 +41,74 @@ internal OpenTelemetryLogger(string categoryName, OpenTelemetryLoggerProvider pr
...
+                };
+
+                var activity = Activity.Current;
+                if (activity != null)
+                {
+                    record.TraceId = activity.TraceId;
+                    record.SpanId = activity.SpanId;
+                    record.TraceState = activity.TraceStateString;
+                    record.TraceFlags = activity.ActivityTraceFlags;
+                }
```
Comment (mic-max) [inline src/OpenTelemetry/Logs/OpenTelemetryLogger.cs:71]: If this logic is or was to be used elsewhere, I'd say it could be a function of `LogRecord`

**Review by mic-max — APPROVED** [review]

File: src/OpenTelemetry/Logs/OpenTelemetryLogger.cs (hunk 1 above)
Comment (reyang) [reply src/OpenTelemetry/Logs/OpenTelemetryLogger.cs:71]: Yeah, something like `LogRecord.Clear()` or `LogRecord.Restock()`?

**Review by Yun-Ting — APPROVED** [review]

**Pull request closed without merging**